packages/maskbook/src/extension/background-script/WelcomeServices/generateBackupJSON.ts

```diff
@@ -102,6 +109,29 @@ export async function generateBackupJSON(opts: Partial<BackupOptions> = {}): Pro
         const wallets_ = (await getWallets(ProviderType.Maskbook)).map(WalletRecordToJSONFormat)
         wallets.push(...wallets_)
     }
+
+    async function backupAllPlugins() {
+        await Promise.all(
+            [...activatedPluginsWorker]
+                // generate backup
+                .map(async (plugin) => {
+                    const f = plugin.backup?.onBackup
+                    if (!f) return
+
+                    async function backupPlugin() {
+                        const object = await timeout(f!(), 3000)
+                        if (object.none) return
+                        // We limit the plugin contributed backups must be simple objects.
+                        // We may allow plugin to store binary if we're moving to binary backup format like messagepack.
+                        plugins[plugin.ID] = object.map(JSON.stringify).map(JSON.parse).val
+                    }
+                    if (process.env.NODE_ENV === 'development') return backupPlugin()
```

Comment:
```suggestion
```

Reply:
This is intended to fail the whole backup progress in the development instead of a error log in the background.

Reply:
What about move this development detecting into `catch()`, and re-throwing the caught error in `catch()`, then we can also find the error from logs.